src/fal/cli/model_generator/module_check.py

```diff
@@ -26,30 +26,52 @@ def generate_dbt_dependencies(module: ast.Module) -> str:
     return "\n".join(map(lambda s: "{{ " + s.strip() + " }}", lines))
 
 
-def _find_function_calls(module: ast.Module) -> List[ast.Call]:
-    return [node for node in ast.walk(module) if isinstance(node, ast.Call)]
+def write_to_model_check(module: ast.Module):
+    """
+    Make sure a there is a single write_to_model function call on the top level
+    """
+
+    all_function_calls = _find_function_calls(ast.walk(module))
+    all_wtm_calls = _filter_function_calls_by_name(all_function_calls, "write_to_model")
+
+    top_level_function_calls = _find_function_calls(
+        stmt.value for stmt in module.body if isinstance(stmt, ast.Expr)
+    )
+    assert (
+        len(all_wtm_calls) == 1 and all_wtm_calls[0] in top_level_function_calls
+    ), "`write_to_model` must be called once on the top level of the Python Model script"
+
 
+def _find_function_calls(nodes: Iterator[ast.AST]) -> List[ast.Call]:
+    return [node for node in nodes if isinstance(node, ast.Call)]
 
-def _find_dbt_function_calls(calls: List[ast.Call], func_name: str) -> List[ast.Call]:
+
+def _filter_function_calls_by_name(calls: List[ast.Call], func_name: str):
     """
-    Analyze all function calls in the file to find the ones that call `func_name` with all literal arguments.
-    We ignore a `ref(var)` but accept a `ref('model_name')`
+    Analyze all function calls passed to find the ones that call `func_name`.
     """
-    func_calls = [
+    return [
         call
         for call in calls
         if isinstance(call.func, ast.Name) and call.func.id == func_name
     ]
 
-    def is_constant(arg: ast.expr):
+
+def _find_dbt_function_calls(calls: List[ast.Call]) -> List[ast.Call]:
```

Comment:
Since this function does not filter by name anymore, should we rename it to something like `_filter_inferable_calls` / `_filter_constant_calls` or something like that?